src/core/public/rendering/rendering_service.tsx

```diff
@@ -39,28 +43,40 @@ interface StartDeps {
  * @internal
  */
 export class RenderingService {
-  start({ chrome, targetDomElement }: StartDeps) {
-    const chromeUi = chrome.getComponent();
-    const legacyRef = React.createRef<HTMLDivElement>();
+  start({ application, chrome, injectedMetadata, targetDomElement }: StartDeps): RenderingStart {
+    const chromeUi = chrome.getHeaderComponent();
+    const appUi = application.getComponent();
+
+    const legacyMode = injectedMetadata.getLegacyMode();
+    const legacyRef = legacyMode ? React.createRef<HTMLDivElement>() : null;
 
     ReactDOM.render(
       <I18nProvider>
         <div className="content" data-test-subj="kibanaChrome">
           {chromeUi}
 
-          <div ref={legacyRef} />
+          {!legacyMode && (
+            <div className="app-wrapper">
```

Comment:
should we add a test to check `app-wrapper` is present in DOM tree? it's a part of public API https://github.com/restrry/kibana/blob/fe51d3419fefc464e8a5672d66bb625d7d5e6b06/x-pack/legacy/plugins/canvas/public/components/popover/popover.js#L67